### PR TITLE
Meeting Card Skeleton

### PR DESCRIPTION
**Pull request by firejake308**
I have written a skeleton for the meeting card and appropriate tests. The tests are pretty simple, as the `MeetingCard` is a pretty simple component, and although I am aware that certain features will have to be added later, such as coloring and positioning, I would like to keep it simple enough to be someone's first React component, if they've never seen React before. I am aware that class components are generally more intuitive, but ESLint demands that stateless components be functional, and it isn't too hard in my opinion to get used to the idea of calling a function with certain arguments that returns a HTML element.

### CODE REVIEW

File: autoscheduler/frontend/src/types.ts [hunk 1]
```diff
@@ -1,3 +1,32 @@
...
+export interface Section {
+  id: number;
+  subject: string;
+  courseNum: number;
+  sectionNum: number;
+  minCredits: number;
+  maxCredits: number | null;
+  currentEnrollment: number;
+  instructor: number;
+}
```
Comment (gannonprudhomme) [inline autoscheduler/frontend/src/types.ts]: Since this is basically the backbone of the front-end, I think this deserves its own PR so we can discuss what all we actually need.

Comment (firejake308) [reply autoscheduler/frontend/src/types.ts]: Do you want all of the types from `scraper/models.py` implemented? I only made the two that I needed to get the Meeting testing working, but it's trivial to copy the types from the backend branch if you want to get it all done at once

Comment (gannonprudhomme) [reply autoscheduler/frontend/src/types.ts]: . Only downside is there may be things we can't forsee that we'll need when actually using them, but I think it'll be fine if we change them in the future

Comment (gannonprudhomme) [reply autoscheduler/frontend/src/types.ts]: I don't expect it to be a 1:1 to the models(i.e. the department probably won't need its term)

**Comment by gannonprudhomme** [comment]
Attached to #47 as there isn't an issue for this, and there really won't be any other outline for the schedule components(as I assume Adel will make the schedule component on his own)

**Comment by gannonprudhomme** [comment]
Can this be remade(or rebased) with the up to date changes that are now on `frontend/master` from #58 ?

File: autoscheduler/frontend/src/.eslintrc.js [hunk 2]
```diff
@@ -45,5 +45,6 @@ module.exports = {
   rules: {
     "react/jsx-filename-extension": [1, {"extensions": [".jsx", ".tsx"]}],
     "import/no-extraneous-dependencies": ["error", {"devDependencies": ["**/*.test.ts", "**/*.test.tsx"]}],
+    "react/prop-types": 0
```
Comment (gannonprudhomme) [inline autoscheduler/frontend/src/.eslintrc.js]: What does this rule specify? Is this related to #53 ?

Comment (firejake308) [reply autoscheduler/frontend/src/.eslintrc.js]: Yes.

File: autoscheduler/frontend/src/tests/Meeting.test.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import { render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import * as React from 'react';
+
+import MeetingCard from '../components/MeetingCard';
+import { Meeting, MeetingType, Section } from '../types';
```
Comment (gannonprudhomme) [inline autoscheduler/frontend/src/tests/Meeting.test.tsx]: Does this not throw an error for you? I get a `Cannot find module '../types'` error when I try to run the tests.

File: autoscheduler/frontend/src/components/MeetingCard.tsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+/* eslint-disable react/prop-types */
```
Comment (gannonprudhomme) [inline autoscheduler/frontend/src/components/MeetingCard.tsx]: Can we remove this line, per your change in `.eslintrc.js`

File: autoscheduler/frontend/src/tests/Meeting.test.tsx (hunk 3 above)
Comment (firejake308) [reply autoscheduler/frontend/src/tests/Meeting.test.tsx]: It does, I'll have to refactor it to reflect the latest commits to frontend/master

File: autoscheduler/frontend/src/components/MeetingCard.tsx (hunk 4 above)
Comment (firejake308) [reply autoscheduler/frontend/src/components/MeetingCard.tsx]: Yes

File: autoscheduler/frontend/src/tests/Meeting.test.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+test('displays subject and course number', () => {
+  const { getByText } = render(<MeetingCard meeting={testMeeting} />);
+  expect(getByText(/CSCE/)).toBeTruthy();
+  expect(getByText(/121/)).toBeTruthy();
+});
+
+test('subject and course number can adapt to random values', () => {
+  // arrange
+  const randSubject = new Array(5).reduce((acc) => acc + String.fromCharCode(
```
[the remark this answers is not in the document]
Comment (gannonprudhomme) [reply autoscheduler/frontend/src/tests/Meeting.test.tsx]: > I'm not sure I like using random values for unit testing, unless we give a seed to the random function (we'd have to use seedrandom in npm or something similar), since it makes tests unpredictable. There isn't a reason that I can think of that using random values would change anything versus just giving it values we made manually, either, since it's just a static length string and a number 100-599

Yeah I agree, there really isn't a logical reason to have it be a random value, since there isn't logic related to the actual value of the subject, as long as it's a certain length. See [Unit Testing Standards](https://github.com/aggie-coding-club/Automatic-Aggie-Scheduler/wiki/Unit-Testing-Standards), specifically the "Readable" section.

**Comment by gannonprudhomme** [comment]
Given our discussion @firejake308 (which I think you were at my place for @ryan-conn ) about displaying the meeting type on the MeetingCard, should there be a test for it? Or not since we haven't determined exactly how it should be displayed.

**Comment by ryan-conn** [comment]
I'll write some tests to make sure it displays correctly while I'm making the actual MeetingCard component.

**Comment by gannonprudhomme** [comment]
> I'll write some tests to make sure it displays correctly while I'm making the actual MeetingCard component.

In that case, should we even merge this into `frontend/master`? You(ryan) could just use it when you're working on it.

That being said, the benefit of merging it to master is that there's less changes to look at when Ryan makes a PR for #54

**Comment by ryan-conn** [comment]
> In that case, should we even merge this into `frontend/master`? You(ryan) could just use it when you're working on it.
> 
> That being said, the benefit of merging it to master is that there's less changes to look at when Ryan makes a PR for #54

Yeah I don't think this should get merged, I think @firejake308 should push `Meeting.test.tsx` (rename to `MeetingCard.test.tsx` and move to `frontend/src/tests/components`) as well as the `eslintrc` change to the `frontend/meeting-card` branch I made a little bit ago and we could just make one pull request for all the changes.
